src/microbe_stage/CompoundCloudPlane.cs

```diff
@@ -400,6 +400,9 @@ public void AddCloud(Compound compound, float density, int x, int y)
     /// <returns>The amount of compound taken</returns>
     public float TakeCompound(Compound compound, int x, int y, float fraction = 1.0f)
     {
+        if (fraction < 0.0f)
+            return 0.0f;
```

Comment:
I think we should rather try to fix the problem at the source, ie. throw an exception here for invalid parameters if fraction is negative.

Reply:
Perhaps, but without the other change below, this will still be hit with fraction negative. So we should not do that for now.

Reply:
As I said on discord, I think this hack is fine to keep in the game for two weeks as long as there's a TODO, which I'm fine with merging at this last minute, but that other change to the 2x vent threshold I wouldn't want to merge.

I wrote some TODO list items for myself about a proper fix (which will need some safety time to verify it works):
- comment on the 2* vent code
- potentially quick fix for negative fraction in absorb, issue to properly change to exception and fix everywhere after release
- create a new "remaining space method" for compounds that has Max(0, Capacity - Amount)